### PR TITLE
Enabling block-by-block evaluation for granite-3.x-models

### PR DESCRIPTION
**Pull request by bayo-ibm**
…te-3-models

### Description of the change
The block-by-block model evaluation does not work with granite-3.x models. This is because the eval_llm_1GPU does not take into consideration the peculiarities associated with granite-3.x-models. 

The quant/ptq.py was updated to reflect the current naming conventions for the various blocks in granite-3.x models.
The eval_llm_1GPU function was updated to correctly calculate the logits for granite-3. x models.
### Related issues or PRs

#155 
### Was the PR tested
The fix was validated by performing FP8 DQ-SQ using the granite-3.0-8B-Instruct/granite-3.3-8B-Instruct models. The evaluation of the quantized and the unquantized models using eval_llm_1GPU achieved the same results as using the "evaluator.evaluate" method

- [ ] I have added >=1 unit test(s) for every new method I have added (if that coverage is difficult, please briefly explain the reason)
- [ ] I have ensured all unit tests pass

### Checklist for passing CI/CD:

- [ ] All commits are signed showing "Signed-off-by: Name \<email@domain.com\>" with `git commit -signoff` or equivalent
- [ ] PR title and commit messages adhere to [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [ x] Contribution is formatted with `tox -e fix`
- [x ] Contribution passes linting with `tox -e lint`
- [x ] Contribution passes spellcheck with `tox -e spellcheck`
- [ x] Contribution passes all unit tests with `tox -e unit`

Note: CI/CD performs unit tests on multiple versions of Python from a fresh install.  There may be differences with your local environment and the test environment.

### CODE REVIEW

File: fms_mo/quant/ptq.py [hunk 1]
```diff
@@ -2140,14 +2140,22 @@ def get_blocks(model, model_type=None):
             None,
             "lm_head",
         ),
-        "granite": (
+        "granite_old": (
```
[the remark this answers is not in the document]
Comment (bayo-ibm) [reply fms_mo/quant/ptq.py]: corrected

File: fms_mo/utils/eval_utils.py [hunk 2]
```diff
@@ -98,6 +98,7 @@ def eval_llm_1GPU(qcfg, model, test_dataset, pre_cache_func=None, **kwargs):  #
     logger.info("All blocks are computed for evaluation")
 
     nlls = []
+    logits_scaling = getattr(model.config, "logits_scaling", 1)
```
Comment (andrea-fasoli) [inline fms_mo/utils/eval_utils.py:101]: I would add a comment along the line of:
```# Granite-3.X (and potentially other models) require scaling of the lm_head outputs to obtain the logits```

**Review by andrea-fasoli — APPROVED** [review]
